### ISSUE TITLE
Why P, R and mAP all suddenly drop to zero during training?

### ISSUE THREAD

**Issue by wyx1997** [ultralytics/yolov5#2037]
## ❔Question
I have trained yolov5x model on my custom data with the default setting. The P, R and mAP all increased normally in the first serveral epochs, but they suddenly droped to zero and remained zero.  I wonder why it happened? And How Can I solve it? Should I reduce the lr?

## Additional context
I got these information only in runs/train/exp/results.txt. Here is the screenshot of results.
![image](https://user-images.githubusercontent.com/29030808/105657715-009f2300-5f00-11eb-883e-a449a8ef0659.png)

**Comment by glenn-jocher** [comment]
👋 Hello, thank you for your interest in our work! This issue seems to lack the minimum requirements for a proper response, or is insufficiently detailed for us to help you. Please note that most technical problems are due to:

- **Your modified or out-of-date code**. If your issue is not reproducible in a new `git clone` version of this repo we can not debug it. Before going further run this code and verify your issue persists:
```bash
$ git clone https://github.com/ultralytics/yolov5 yolov5_new  # clone latest
$ cd yolov5_new
$ python detect.py  # verify detection

# CODE TO REPRODUCE YOUR ISSUE HERE
```

- **Your custom data.** If your issue is not reproducible in one of our 3 common datasets ([COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [COCO128](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), or [VOC](https://github.com/ultralytics/yolov5/blob/master/data/voc.yaml)) we can not debug it. Visit our [Custom Training Tutorial](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data) for guidelines on training your custom data. Examine `train_batch0.jpg` and `test_batch0.jpg` for a sanity check of your labels and images.

- **Your environment.** If your issue is not reproducible in one of the verified environments below we can not debug it. If you are running YOLOv5 locally, verify your environment meets all of the [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies specified below. If in doubt, download Python 3.8.0 from https://www.python.org/, create a new [venv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/), and install requirements.

If none of these apply to you, we suggest you close this issue and raise a new one using the 🐛 **Bug Report template**, providing screenshots and **minimum viable code to reproduce your issue**. Thank you!


## Requirements

Python 3.8 or later with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed, including `torch>=1.7`. To install run:
```bash
$ pip install -r requirements.txt
```


## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab Notebook** with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
- **Kaggle Notebook** with free GPU: [https://www.kaggle.com/models/ultralytics/yolov5](https://www.kaggle.com/models/ultralytics/yolov5)
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) 
- **Docker Image** https://hub.docker.com/r/ultralytics/yolov5. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) ![Docker Pulls](https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker)

## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu.

**Comment by ycdhqzhiai** [comment]
@wyx1997 I've had the same problem.At first, I thought it was a problem with my data or code version, but I checked my data and changed the code version, but the problem was still not solved

**Comment by wyx1997** [comment]
@ycdhqzhiai Hi, I encountered this problem today. I firstly trained yolov5x model with default setting but lr was set to 0.015, but the results were even worse than yolov5s. Then I set lr to defalut value(0.01), and the problem above occured. Have you ever tried different lr cause I think it may be important for this problem? Or there may be a bug here?

**Comment by ycdhqzhiai** [comment]
@wyx1997 I haven't tried any other learning rate.I don't think it's a bug because I trained normally a month ago.But now the map will be reduced to 0(results are shown in the figure).The only difference is using different servers(2080ti and 3090).I don't have a 2080ti server right now so I can't reproduce the previous results

![image](https://user-images.githubusercontent.com/12273405/105691398-26481e80-5f38-11eb-9e15-751430afe0e0.png)

**Comment by wyx1997** [comment]
@ycdhqzhiai It seems that mAP in your experiment bounces after droping to zero. But in my results, it remains zero for serveral epochs. So I think the problems of our experiment may be different. I want to try some lower learning rates and try on different servers. Anyway, thank you for your share!

**Comment by glenn-jocher** [comment]
@wyx1997 your training losses are unstable, they are increasing rather than decreasing, indicating dataset problems or inappropriate hyperparameters, and the same applies to @ycdhqzhiai.

If this is reproducible in a common dataset (COCO, COCO128, VOC) in a common environment (Colab, Docker, Kaggle etc) then please let us know, otherwise the issue is likely in your code/hyperparameter modifications or your dataset (image/label issues).

**Comment by wyx1997** [comment]
@ycdhqzhiai OK, thank you very much! I will check my dateset again and try to adjust hyperparameters.
